### PR TITLE
test(client): mongo query conflict #14001

### PR DESCRIPTION
**Pull request by millsp**

### CODE REVIEW

**Comment by danstarns** [comment]
Related: https://github.com/prisma/prisma/issues/14001

Related: https://github.com/prisma/prisma-engines/pull/3185

File: packages/client/tests/functional/issues/14001-mongo-order-by-conflict/tests.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      await expect(result).resolves.toMatchInlineSnapshot(`null`)
+    })
+
+    test('findMany', async () => {
+      const result = prisma.resource.findMany({
+        where: { OrderBy: { gt: 0 } },
+        orderBy: { OrderBy: 'asc' },
+        cursor: { OrderBy: 1 },
+        distinct: ['OrderBy'],
+      })
```
Comment (janpio) [inline packages/client/tests/functional/issues/14001-mongo-order-by-conflict/tests.ts:30]: The chose nquery here is different than the original reproduction from the user: https://bitbucket.org/awimmjsystems/prisma/src/de00769d30bb07c78f797b2dfe6e1d4adaeef982/index.js#lines-8:11 Intentional or confirmed irrelevant?

Comment (millsp) [reply packages/client/tests/functional/issues/14001-mongo-order-by-conflict/tests.ts:30]: I confirmed that it was indeed failing first with this test, then made sure it does not fail on other levels and across other kinds of queries just to be more thorough.

**Comment by janpio** [comment]
I confirmed manually that this fails with engines version `4.4.0-18.dc7a16204a5f0f2d962929cb297b3f4b3766d689` (which is 1 before the fix got merged afaik).

File: packages/client/tests/functional/issues/14001-mongo-order-by-conflict/tests.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+      await expect(result).resolves.toMatchInlineSnapshot(`Array []`)
```
Comment (danstarns) [inline packages/client/tests/functional/issues/14001-mongo-order-by-conflict/tests.ts:32]: Wondering if we should be asserting that, (1) the query doesn't crash, (2) that the data returned is correct...

Thinking we should add some dummy data in the beforeAll hook and assert the response of these queries.

**Review by danstarns — APPROVED** [review]

File: packages/client/tests/functional/issues/14001-mongo-order-by-conflict/tests.ts (hunk 2 above)
Comment (millsp) [reply packages/client/tests/functional/issues/14001-mongo-order-by-conflict/tests.ts:32]: Yes, I think having dummy data does not hurt, I'll add.

Comment (millsp) [reply packages/client/tests/functional/issues/14001-mongo-order-by-conflict/tests.ts:32]: Turns out it uncovered something surprising. I will merge PR as-is, and report the crash in a new issue.

Comment (millsp) [reply packages/client/tests/functional/issues/14001-mongo-order-by-conflict/tests.ts:32]: I got confused, this is the one that I tried to add dummy data to and it crashed https://github.com/prisma/prisma/pull/15215 I reported the issue already

**Pull request merged**